### ISSUE TITLE
使用vlm-transformer推理 报warning: pad token is same as eos token

### ISSUE THREAD

**Issue by chong-w** [opendatalab/MinerU#3307]
### 🔎 Search before asking | 提交之前请先搜索

- [x] I have searched the MinerU [Readme](https://github.com/opendatalab/MinerU) and found no similar bug report.
- [x] I have searched the MinerU [Issues](https://github.com/opendatalab/MinerU/issues) and found no similar bug report.
- [x] I have searched the MinerU [Discussions](https://github.com/opendatalab/MinerU/discussions) and found no similar bug report.

### 🤖 Consult the online AI assistant for assistance | 在线 AI 助手咨询

- [x] I have consulted the [online AI assistant](https://deepwiki.com/opendatalab/MinerU) but was unable to obtain a solution to the issue.

### Description of the bug | 错误描述

warning: The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results

minerU commit id:  be4f3de32b58ccf81c6a6dcb9d3e4998424cee6a
transformers版本: 4.55.0

### How to reproduce the bug | 如何复现

(base) ubuntu:~/MinerU$ mineru -p /home/wangc/MinerU/pdf -o output  -b vlm-transformers -d cuda:0
2025-08-13 21:55:54.179 | WARNING  | mineru.backend.vlm.predictor:<module>:35 - sglang is not installed. If you are not using sglang, you can ignore this warning.
Downloading Model from https://www.modelscope.cn to directory: /home/wangc/.cache/modelscope/hub/models/OpenDataLab/MinerU2.0-2505-0.9B
2025-08-13 21:55:55,952 - modelscope - INFO - Target directory already exists, skipping creation.
/home/wangc/miniconda3/lib/python3.13/site-packages/accelerate/utils/modeling.py:1614: UserWarning: The following device_map keys do not match any submodules in the model: ['model.image_newline']
  warnings.warn(
2025-08-13 21:55:57.419 | INFO     | mineru.backend.vlm.predictor:get_predictor:110 - get_predictor cost: 1.47s
Predict:   0%|                                                                                                                                                                            | 0/8 [00:00<?, ?it/s]The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Predict: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [01:16<00:00,  9.62s/it]

### Operating System Version| 操作系统版本

ubuntu 24.04

### Python version | Python 版本

3.13

### Software version | 软件版本 (mineru --version)

2.0.x

### Device mode | 设备模式

cuda

**Comment by renatilwp** [comment]
在mineru/backend/vlm/hf_predictor.py第140-151行修改：
       inputs = self.tokenizer(prompt, return_tensors="pt", padding=True)
        input_ids = inputs.input_ids.to(self.model.device)
        attention_mask = inputs.attention_mask.to(self.model.device)
        output_ids = self.model.generate(
            input_ids,
            attention_mask=attention_mask,
            images=image_tensor,
            image_sizes=image_sizes,
            use_cache=True,
            **generate_kwargs,
            **kwargs,
        )

**Comment by renatilwp** [comment]
Modify lines 140-151 of /mineru/backend/vlm/hf_predictor.py to:
     inputs = self.tokenizer(prompt, return_tensors="pt", padding=True)
        input_ids = inputs.input_ids.to(self.model.device)
        attention_mask = inputs.attention_mask.to(self.model.device)
        output_ids = self.model.generate(
            input_ids,
            attention_mask=attention_mask,
            images=image_tensor,
            image_sizes=image_sizes,
            use_cache=True,
            **generate_kwargs,
            **kwargs,
        )
